controller_manager/src/controller_manager.cpp

```diff
@@ -2366,19 +2366,62 @@ controller_interface::return_type ControllerManager::update(
         run_controller_at_cm_rate ? period
                                   : rclcpp::Duration::from_seconds((1.0 / controller_update_rate));
 
-      const rclcpp::Time current_time = get_clock()->now();
+      rclcpp::Time current_time;
+      if (get_clock()->started())
+      {
+        current_time = get_clock()->now();
+      }
+      else if (
+        time == rclcpp::Time(0, 0, this->get_node_clock_interface()->get_clock()->get_clock_type()))
+      {
+        throw std::runtime_error(
+          "No clock received, and time argument is zero. Check your controller_manager node's "
+          "clock configuration (use_sim_time parameter) and if a valid clock source is "
+          "available. Also pass a proper time argument to the update method.");
+      }
+      else
+      {
+        // this can happen with use_sim_time=true until the /clock is received
+        RCLCPP_WARN(
+          get_logger(),
+          "No clock received, using time argument instead! Check your node's clock "
+          "configuration (use_sim_time parameter) and if a valid clock source is available.");
```

Comment:
Just for the sake of completeness: The following would work.
```suggestion
rclcpp::Clock clock = rclcpp::Clock();
          RCLCPP_WARN_THROTTLE(
            get_logger(),
            clock,
            1000,
            "No clock received, using time argument instead! Check your node's clock "
            "configuration (use_sim_time parameter) and if a valid clock source is available");
```

Reply:
you are totally right!